src/main/java/com/digitalasset/testing/junit4/Sandbox.java

```diff
@@ -234,140 +164,26 @@ public ExternalResource getRule() {
       @Override
       protected void before() throws Throwable {
         if (useReset) {
-          startCommChannels(sandboxPort);
+          sandboxCommunicator.startCommChannels();
```

Comment:
All this logic should be in SandboxCommunicator. Its API should look like this:
```
void start()
void stop()
void restart()
void reset()
```

Reply:
Done.